project-icp/bee-pollinator-app
# Add new user tour

## Overview

Add a tour to show new users around the model/compare stages of the app. 

We're using the tour library [hopscotch](http://linkedin.github.io/hopscotch/). It has its own API to handle state of seen/not seen using session storage, but it has several problems:
 - it only expects to be managing a single tour, and the tour for this app has three stages: when you're on the current crop map tab, when you're on "new scenario" and when you're in the compare view. These three stages were best implemented using three separate tours
- it resets the state after a tour has ended so you can't tell if you've never seen a tour or if you've completed one

Because of this I've mostly ignored what hopscotch thinks the state is and wrapped each tour object in our own object that manages state via local storage (and whether there's a user logged in with more than one project). 

See wireframes for desired text and placement https://marvelapp.com/h3g8796/screen/28035773

@jfrankl may also want a look at this 

Connects #217 

## Demo
<img width="629" alt="screen shot 2017-07-07 at 8 13 06 pm" src="https://user-images.githubusercontent.com/7633670/27980822-ba1e2736-6350-11e7-80c2-a46dc3c73c66.png">

## Testing
- Pull branch and `./scripts/npm.sh install` to get hopscotch, then `./scripts/bundle.sh --vendor`
- As a guest user, progress to the model, confirm you see the initial tour bubble over current crop map
- Go through part of the tour, and then refresh the page or navigate back to draw
- Confirm the tour is right where you left it when you return
- Complete the tour and confirm it no longer appears on refresh
- Enter `localStorage.clear()` in the js console (the tour should reappear on subsequent refreshes)
- Login to an account with more than one project and confirm the tour doesn't appear
- Click around in any other order you can think ...
